cpp/CMakeLists.txt

```diff
@@ -34,7 +34,7 @@ else()
     else()
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
     endif()
-    set(CMAKE_CXX_FLAGS_DEBUG "-ggdb")
+    set(CMAKE_CXX_FLAGS_DEBUG "-g3 -fsanitize=address -fsanitize=undefined -fsanitize=float-divide-by-zero -fsanitize=float-cast-overflow")
```

Comment:
Can we hide this behind an option/cached variable so that it's not on at all times and users can decide when to turn it on/off

Reply:
Yes that is the plan (this PR is WIP).